lib/AST/DeclContext.cpp

```diff
@@ -476,20 +476,15 @@ unsigned DeclContext::getSemanticDepth() const {
 }
 
 bool DeclContext::mayContainMembersAccessedByDynamicLookup() const {
-  // Dynamic lookup can only find class and protocol members, or extensions of
-  // classes.
-  if (auto *NTD = dyn_cast<NominalTypeDecl>(this)) {
-    if (isa<ClassDecl>(NTD))
-      if (!isGenericContext())
-        return true;
-    if (auto *PD = dyn_cast<ProtocolDecl>(NTD))
-      if (PD->getAttrs().hasAttribute<ObjCAttr>())
-        return true;
-  } else if (auto *ED = dyn_cast<ExtensionDecl>(this)) {
-    if (auto *CD = dyn_cast<ClassDecl>(ED->getExtendedNominal()))
-      if (!CD->isGenericContext())
-        return true;
-  }
+  // Members of non-generic classes and class extensions can be found by
+  /// dynamic lookup.
+  if (auto *CD = getSelfClassDecl())
+    return !CD->isGenericContext();
```

Comment:
Oops. Generic classes have been able to have `@objc` members for a while. Guess we never updated this (and probably shouldn't at this point).

Reply:
You're right. I preserved this behavior without even thinking about it.